src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -420,18 +427,23 @@ export const selectPortfolioAccountRows = createDeepEqualOutputSelector(
         const precision = assetsById[assetId]?.precision
         const price = marketData[assetId]?.price
         const cryptoAmount = fromBaseUnit(baseUnitBalance, precision)
-        const fiatAmount = bnOrZero(cryptoAmount).times(bnOrZero(price)).toFixed(2)
-        const allocation = bnOrZero(fiatAmount)
-          .div(bnOrZero(totalPortfolioFiatBalance))
-          .times(100)
-          .toNumber()
+        const fiatAmount = bnOrZero(cryptoAmount).times(bnOrZero(price))
+        /**
+         * if fiatAmount is less than the selected threshold,
+         * continue to the next asset balance by returning acc
+         */
+        if (fiatAmount.isLessThan(bnOrZero(balanceThreshold))) {
+          return acc
+        }
+        const allocation = fiatAmount.div(bnOrZero(totalPortfolioFiatBalance)).times(100).toNumber()
         const priceChange = marketData[assetId]?.changePercent24Hr
         const data = {
           assetId,
           name,
           icon,
           symbol,
-          fiatAmount,
+          // second parameter is for rounding down instead of up
+          fiatAmount: fiatAmount.toFixed(2, 1),
```

Comment:
can we please move this logic to the line above so we're just assembling an object to return